Separate tips for GDB, Pwndbg and LLDB

may3rr opened the pull request:
**TLDR**: Split tips into GDB, Pwndbg, and LLDB categories to show appropriate tips based on debugger type.

Resolves #2768

This PR separates the existing tips into three distinct categories:
- GDB-specific tips
- Pwndbg-specific tips 
- LLDB-specific tips (newly added)

Changes made:
1. Split the original `TIPS` list into `GDB_TIPS`, `PWNDBG_TIPS`, and `LLDB_TIPS`
2. Added 20 new LLDB-specific tips
3. Modified the `get_tip_of_the_day()` function to accept a `debugger_type` parameter that determines which tips to show based on the environment (GDB or LLDB)
4. Maintained backward compatibility by keeping the original `TIPS` list as a combination of GDB and Pwndbg tips

This allows the tips to be properly displayed based on whether the user is in a GDB or LLDB environment.

may3rr committed:
Separate tips for GDB, Pwndbg and LLDB

may3rr committed:
Fix linting issues

may3rr committed:
Fix docstring formatting

disconnect3d commented:
Hi, thanks! This makes sense, although I don't think we need any "backward compatibility" here. We can just drop the "TIPS" variable.

may3rr committed:
Remove redundant TIPS variable as suggested

disconnect3d commented:
The commands tips needs to be updated as well:

```
Traceback (most recent call last):
  File "/nix/store/14[7](https://github.com/pwndbg/pwndbg/actions/runs/13699724159/job/38310231003?pr=2773#step:5:8)sm174spmwwiqifp9zp3p843wb8i15-pwndbg/share/pwndbg/gdbinit.py", line 223, in <module>
    main()
  File "/nix/store/147sm174spmwwiqifp9zp3p[8](https://github.com/pwndbg/pwndbg/actions/runs/13699724159/job/38310231003?pr=2773#step:5:9)43wb8i15-pwndbg/share/pwndbg/gdbinit.py", line 206, in main
    pwndbg.dbg.setup()
  File "/nix/store/147sm174spmwwiqifp[9](https://github.com/pwndbg/pwndbg/actions/runs/13699724159/job/38310231003?pr=2773#step:5:10)zp3p843wb8i15-pwndbg/share/pwndbg/pwndbg/dbg/gdb/__init__.py", line 1273, in setup
    load_commands()
  File "/nix/store/147sm174spmwwiqifp9zp3p843wb8i15-pwndbg/share/pwndbg/pwndbg/commands/__init__.py", line 790, in load_commands
    import pwndbg.commands.tips
  File "/nix/store/147sm174spmwwiqifp9zp3p843wb8i15-pwndbg/share/pwndbg/pwndbg/commands/tips.py", line 6, in <module>
    from pwndbg.lib.tips import TIPS
ImportError: cannot import name 'TIPS' from 'pwndbg.lib.tips' (/nix/store/[14](https://github.com/pwndbg/pwndbg/actions/runs/13699724159/job/38310231003?pr=2773#step:5:15)7sm174spmwwiqifp9zp3p843wb8i[15](https://github.com/pwndbg/pwndbg/actions/runs/13699724159/job/38310231003?pr=2773#step:5:16)-pwndbg/share/pwndbg/pwndbg/lib/tips.py)
```

And I believe there is some standard way to determine the debugger type, but I don't have time to look at this right now.

Maybe @patryk4815 knows

patryk4815 commented:
> Maybe @patryk4815 knows

I think right now only option:
`pwndbg.dbg.is_gdblib_available()` if `True` - gdb
`pwndbg.dbg.is_gdblib_available()` if `False` - lldb

Another option could be to implement a new method in`pwndbg.dbg` eg. `pwndbg.dbg.type()` 🤔

may3rr committed:
Fix: Update tips command to work with separated debugger tips

- Removed TIPS variable for backward compatibility
- Added get_all_tips() function to retrieve appropriate tips based on debugger
- Updated tips.py to use get_all_tips() instead of importing TIPS
- Modified get_tip_of_the_day() to use pwndbg.dbg.is_gdblib_available()

may3rr committed:
fix lint

may3rr commented:
Hey there! I've updated the PR based on your feedback. Here's what I did:

1. Removed the TIPS variable(following your suggestion @disconnect3d 
2. Added a `get_all_tips()` function to get the appropriate tips based on the debugger type
3. Updated the tips.py command to use this function instead of importing TIPS directly
4. Used `pwndbg.dbg.is_gdblib_available()` to detect the debugger type as suggested by @patryk4815 

BTW this is my first PR, so thanks a lot for your guidance and patience!

may3rr committed:
fix lint

peace-maker commented on pwndbg/lib/tips.py:
```diff
@@ -51,9 +54,51 @@
+    Returns a random tip based on the current debugger type.
+    """
+    import pwndbg.dbg
+    if pwndbg.dbg.is_gdblib_available():
```
This can now use the other function to simplify

```python
return choice(get_all_tips())
```

may3rr committed:
Simplify get_tip_of_the_day() by using get_all_tips()

disconnect3d committed:
Update tips.py

disconnect3d committed:
Update tips.py fix lint
